tests/tcg/arc64/check_qmach.S

```diff
@@ -0,0 +1,553 @@
+; --------------------------------
+; set n flag
+; ----------------
+;  0 * 4 + 1 = 1 and n flag unset
+; ----------------
+; conditions are met, n flag should not be set
+;
+prep_test_case
+  set_reg64 r58, hi=0x00000000, lo=0x00000001
+  set_reg64  r7, hi=0x00000000, lo=0x00000000
+  set_n_flag
+  qmach.n.f    r7, r7, 0
+  flg_none_set
+  chk_reg64  r7, hi=0x00000000, lo=0x00000001
+
+; Test case 19
+; check 0 as register and .f flag
+; (  0,  0,  0,  0) * r1
+; (  0,  0,  0,  0)   s12
+; (     0x0000_0000_0000_0001    )   r58 (acc)
+; --------------------------------
+; set n flag
+; ----------------
+;  0 * 4 + 1 = 1 and n flag unset
+; ----------------
+; n flag should not be set anymore
+;
+prep_test_case
+  set_reg64 r58, hi=0x00000000, lo=0x00000001
+  set_reg64  r1, hi=0x00000000, lo=0x00000000
+  set_n_flag
+  qmach.f    0, r1, 0
+  flg_none_set
+
+; Test case 20
+; check basic limm (limm is duplicated)
+; (     1,      1,      1,      1) * r0
+; (0x1000, 0x0001, 0x1000, 0x0001)   limm
+; (     0x0000_0000_0000_0010    )   r58 (acc)
+; --------------------------------
+;  0x1000 * 2 + 0x0001 * 2 + 0x0000000000000010 = 0x2012
+; ----------------
+;
+prep_test_case
+  set_reg64 r58, hi=0x00000000, lo=0x00000010
+  set_reg64  r0, hi=0x00010001, lo=0x00010001
+  qmach    r0, r0, 0x10000001
+  flg_none_set
+  chk_reg64  r0, hi=0x00000000, lo=0x00002012
+
+; Test case 21
+; check double limm (limm is duplicated)
+; (0x1000, 0x0001, 0x1000, 0x0001)   limm
+; (0x1000, 0x0001, 0x1000, 0x0001)   limm
+; (     0x0000_0000_0000_0010    )   r58 (acc)
+; --------------------------------
+;  0x1000 * 0x1000 * 2 + 0x0001 * 0x0001 * 2 + 0x0000000000000010
+; ----------------
+;  0x0000000002000012
+;
+prep_test_case
+  set_reg64 r58, hi=0x00000000, lo=0x00000010
+  qmach    r0, 0x10000001, 0x10000001
+  flg_none_set
+  chk_reg64  r0, hi=0x00000000, lo=0x02000012
+
+; Test case 22
+; 0 target register with 2 numbers, set flags (only use for this encoding)
+; (0x1000, 0x0001, 0x1000, 0x0001)   limm
+; (0x1000, 0x0001, 0x1000, 0x0001)   limm
+; (     0x0000_0000_0000_0010    )   r58 (acc)
+; --------------------------------
+;  0x1000 * 0x1000 * 2 + 0x0001 * 0x0001 * 2 + 0x0000000000000010
+; ----------------
+;  0x0000000002000012
+;
+prep_test_case
+  set_reg64 r58, hi=0x7fffffff, lo=0xffffffff
+  qmach.f    0, 1, 1
+  flg_nv_set
+
+; Test case 23
+; Check if qmach unsets n flag but doesnt unset the v flag
+; (0x0000, 0x0000, 0x0000, 0x0000)   limm
+; (0x0000, 0x0000, 0x0000, 0x0000)   limm
+; (     0x0000_0000_0000_0000    )   r58 (acc)
+; --------------------------------
+;  0x0000 * 4 + 0x0000000000000000
+; ----------------
+;  0, n flag cleared but v flag untouched
+;
+ prep_test_case
+   set_reg64 r58, hi=0x00000000, lo=0x00000000
+   set_nv_flag
+   qmach.f    0, 0, 0
+   flg_v_set
+
+
+;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;; Reporting ;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;;
+
+valhalla:
+  print "\n[PASS] qmach: qmach qmachu\n"
+  mov r1, 0xdecaf
```

Comment:
this is for tracing in nSIM. please don't forget to remove this line for the final PR.